### PR TITLE
FSE - Show loading indicator on ever navigation to template editor

### LINKED ISSUE #35422: FSE: Loading message for Editor instance.
Via @ianstewart

Loading an editor instance is slow. Even on fast internet. The template part loading screen could be perhaps be improved with a message on the loading screen letting users know we're loading a new screen.

### PR DESCRIPTION
**Pull request by glendaviesnz**
#### Changes proposed in this Pull Request

* Adds the loading indicator to the template edit button on every navigation instead of just on page saves
* Add message as well as loading indicator

#### Testing instructions

* Apply the patch to FSE test env
* Click on Edit Header and Edit Footer buttons and ensure loading indicator and correct message ('Loading Header Editor' for Header and 'Loading Footer Editor' for Footer) shows on button - simulate slow network connection if needed.

**Before**

before-nav

**After**

header-after

Fixes #35422

### CODE REVIEW

File: apps/full-site-editing/full-site-editing-plugin/full-site-editing/blocks/template/edit.js [hunk 1]
```diff
@@ -156,7 +156,13 @@ const TemplateEdit = compose(
 						{ isSelected && (
 							<Placeholder className="template-block__overlay">
 								<Button href={ editTemplateUrl } onClick={ save } isDefault ref={ navButton }>
-									{ navigateToTemplate ? <Spinner /> : sprintf( __( 'Edit %s' ), templateTitle ) }
+									{ navigateToTemplate ? (
```
Comment (glendaviesnz) [inline apps/full-site-editing/full-site-editing-plugin/full-site-editing/blocks/template/edit.js]: I am sure there is a more compact way of doing the following without a fragment - but I couldn't find any other combination that would play nice with the ternary - let me know if there is a better way

Comment (gwwar) [reply apps/full-site-editing/full-site-editing-plugin/full-site-editing/blocks/template/edit.js]: If this gets too big, a helper function is fine to use too

**Comment by Copons** [comment]
This works well, and the ternary is clear enough to me.
But, what do you think about moving the loading message outside of the button?
E.g.
```jsx
<Placeholder>
  { navigateToTemplate && <div><Spinner /> Loading</div> }
  { ! navigateToTemplate && <Button>Edit</Button>
</Placeholder>
```

**Comment by glendaviesnz** [comment]
> But, what do you think about moving the loading message outside of the button?

Yeh, I think that looks a bit nicer, so have done that, but had to hide button with css instead as this is clicked again later in the process to avoid a WP.com issue - https://github.com/Automattic/wp-calypso/blob/279c13d3df1d9fbc47992c0ff321d1327273232d/apps/full-site-editing/full-site-editing-plugin/full-site-editing/blocks/template/edit.js#L108

**Pull request merged**